#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
  ERP 0
  basicTimeStep 50
  FPS 20
  optimalThreadCount 12
  physicsDisableTime 0
}
Viewpoint {
  orientation -0.5645895254853835 0.5506126641449653 0.6148693859636964 1.981493508405472
  position -1.2207409128282443 -5.0903866506046205 36.163552057343765
  follow "robot1"
  followType "None"
}
TexturedBackground {
  texture "factory"
  skybox FALSE
}
TexturedBackgroundLight {
  texture "empty_office"
  castShadows FALSE
}
DEF Robot1 Robot {
  translation -10.349405436772233 -0.9442094666674465 0.07592263755334701
  rotation 4.917789742306367e-17 4.962469104094713e-17 -1 1.5009831567151224
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/1B.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    Solid {
      rotation 0 1 0 0
      children [
        DEF Robot1Body Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "robot1"
  boundingObject USE Robot1Body
  physics Physics {
  }
  controller "RobotControllerV5"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
DEF Robot2 Robot {
  translation -8.875608831992482 0.8002250266898748 0.07592263755334697
  rotation 5.220322824556939e-17 7.352458211424445e-17 -1 1.7104226669544433
  children [
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/2B.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    Solid {
      rotation 0 1 0 0
      children [
        DEF Robot2Body Shape {
          appearance PBRAppearance {
            baseColor 0.25 0.25 0.25
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "robot2"
  boundingObject USE Robot2Body
  physics Physics {
  }
  controller "RobotControllerV5"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 1
}
DEF Robot3 Robot {
  translation -10.688563916870274 -3.7062635486242437 0.07592263755334698
  rotation -3.665906184450794e-17 1.0821652698767376e-17 -1 1.5560109906460213
  children [
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/3B.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    Solid {
      rotation 0 1 0 0
      children [
        DEF Robot3Body Shape {
          appearance PBRAppearance {
            baseColor 0.25 0.25 0.25
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "robot3"
  boundingObject USE Robot3Body
  physics Physics {
    damping Damping {
      linear 0
      angular 0
    }
  }
  controller "RobotControllerV5"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 1
}
DEF Robot4 Robot {
  translation -1.190973552979915 5.162407832426061 0.07592263755334695
  rotation -3.9467936940803614e-17 -9.125917175319607e-18 1 -2.369919849706255
  children [
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/4B.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    Solid {
      rotation 0 1 0 0
      children [
        DEF Robot4Body Shape {
          appearance PBRAppearance {
            baseColor 0.25 0.25 0.25
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "robot4"
  boundingObject USE Robot4Body
  physics Physics {
    damping Damping {
      linear 0
      angular 0
    }
  }
  controller "RobotControllerV5"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 1
}
DEF Robot5 Robot {
  translation -4.232731327157291 -1.0427435739016637 0.07592263755334697
  rotation 4.025242699066254e-18 -3.0774619955947615e-17 0.9999999999999999 -2.3359190614283794
  children [
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/5B.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    Solid {
      rotation 0 1 0 0
      children [
        DEF Robot5Body Shape {
          appearance PBRAppearance {
            baseColor 0.25 0.25 0.25
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "robot5"
  boundingObject USE Robot5Body
  physics Physics {
    damping Damping {
      linear 0
      angular 0
    }
  }
  controller "RobotControllerV5"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 1
}
Robot {
  name "MainController"
  controller "MainAsyncController"
  supervisor TRUE
  synchronization FALSE
  cpuConsumption 0
}
DEF Bola Solid {
  translation -10.5 -6.75 0.0989727
  rotation 0.867712091617179 -0.2829128827309751 0.4087004120944706 -2.6944855450529634
  children [
    DEF Connector Connector {
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0
      axisTolerance 0
      rotationTolerance 0
    }
    DEF Bola_Body Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
        IBLStrength 10
      }
      geometry Sphere {
        radius 0.1
      }
      castShadows FALSE
    }
  ]
  boundingObject USE Bola_Body
  physics Physics {
    density 100
    damping Damping {
    }
  }
}
DEF Baliza1 Transform {
  translation -11.25 0 0
  rotation 0 0 1 3.14
  children [
    DEF Goal1 Solid {
      translation -0.25 1.275 0.51
      children [
        DEF Goal1 Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.1 1.125
          }
        }
      ]
      name "solid(2)"
      boundingObject USE Goal1
    }
    DEF Goal2 Solid {
      translation -0.25 -1.275 0.51
      children [
        DEF Goal2 Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.1 1.125
          }
        }
      ]
      name "solid(3)"
      boundingObject USE Goal2
    }
    DEF Goal0 Solid {
      translation 0 0 0.5075
      children [
        DEF Goal0 Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 2.65 1.125
          }
        }
      ]
      name "solid(1)"
      boundingObject USE Goal0
    }
  ]
}
DEF Baliza2 Transform {
  translation 11.25 0 0
  children [
    DEF Goal1 Solid {
      translation -0.25 1.275 0.51
      children [
        DEF Goal1 Shape {
          appearance PBRAppearance {
            baseColor 1 0.25 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.1 1.125
          }
        }
      ]
      name "Goal1_1"
      boundingObject USE Goal1
    }
    DEF Goal2 Solid {
      translation -0.25 -1.275 0.51
      children [
        DEF Goal2 Shape {
          appearance PBRAppearance {
            baseColor 1 0.25 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.1 1.125
          }
        }
      ]
      name "Goal2_1"
      boundingObject USE Goal2
    }
    DEF Goal0 Solid {
      translation 0 0 0.5075
      children [
        DEF Goal0 Shape {
          appearance PBRAppearance {
            baseColor 1 0.25 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 2.65 1.125
          }
        }
      ]
      name "Goal0_1"
      boundingObject USE Goal0
    }
  ]
}
Robot {
  children [
    Display {
      children [
        RectangleArena {
          rotation 0 1 0 0
          floorSize 24 16
          floorTileSize 24 16
          floorAppearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/Campo.png"
              ]
              repeatS FALSE
              repeatT FALSE
            }
          }
          wallThickness 0.2
          wallHeight 0.5
          wallAppearance PBRAppearance {
            baseColor 0.5 0.5 0.5
            roughness 1
            metalness 0
          }
        }
      ]
      width 240
      height 160
    }
  ]
  name "Display"
  controller "Display"
  supervisor TRUE
  synchronization FALSE
}
DEF Opo1 Robot {
  translation 10.000000419375073 -1.9470072983154547e-06 0.07592263755334705
  rotation 4.7292887188350655e-17 1.9917735870537933e-18 1 -2.857761180032809
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/1R.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    DEF Opo1 Solid {
      rotation 0 1 0 0
      children [
        DEF Opo1Body Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "Opo1"
  boundingObject USE Opo1Body
  physics Physics {
  }
  controller "OpoController"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
DEF Opo2 Robot {
  translation -10.942487752283386 -7.457127598044648 0.07592263755334698
  rotation -5.45278813878001e-17 1.623295397078956e-16 1 0.9282403146649978
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/2R.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    DEF Opo1 Solid {
      rotation 0 1 0 0
      children [
        DEF Opo1Body Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "Opo2"
  boundingObject USE Opo1Body
  physics Physics {
  }
  controller "OpoController"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
DEF Opo3 Robot {
  translation -6.472711823756021 1.9829725316872306 0.07592263755334697
  rotation -5.3866764887540955e-17 -7.090642446099953e-17 1 -2.1070983381866855
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/3R.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    DEF Opo1 Solid {
      rotation 0 1 0 0
      children [
        DEF Opo1Body Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "Opo3"
  boundingObject USE Opo1Body
  physics Physics {
  }
  controller "OpoController"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
DEF Opo4 Robot {
  translation -7.251284454351018 -4.421725520223168 0.075922637553347
  rotation 7.268480748101249e-17 3.1542939886159576e-18 1 -2.567663454210967
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/4R.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    DEF Opo1 Solid {
      rotation 0 1 0 0
      children [
        DEF Opo1Body Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "Opo4"
  boundingObject USE Opo1Body
  physics Physics {
  }
  controller "OpoController"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
DEF Opo5 Robot {
  translation -3.5319728121364538 -2.624234531265768 0.07592263755334692
  rotation -3.1435304955182143e-17 -1.3347145618718005e-17 1 -2.665327957795935
  children [
    Transform {
      translation 0 0 0.055
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "../../OtherFiles/5R.png"
              ]
              filtering 1
            }
          }
          geometry Cylinder {
            height 0.1
            radius 0.24
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
    DEF Connector Connector {
      translation 0.35 0 0
      model "magnetic"
      autoLock TRUE
      distanceTolerance 0.2
      axisTolerance 3.14
      rotationTolerance 3.14
      snap FALSE
    }
    DEF Opo1 Solid {
      rotation 0 1 0 0
      children [
        DEF Opo1Body Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.25
            subdivision 6
          }
          castShadows FALSE
        }
      ]
    }
  ]
  name "Opo5"
  boundingObject USE Opo1Body
  physics Physics {
  }
  controller "OpoController"
  controllerArgs [
    ""
  ]
  supervisor TRUE
  cpuConsumption 0
}
